### PR TITLE
Improve error output

### LINKED ISSUE #1359: Improve error information, include exact value and column
It would be nice quality of life feature if we could return more detailed error info. Including exact value/column which raised it.
Something similar to https://github.com/django-import-export/django-import-export/issues/1215

End goal would be for [row_errors](https://github.com/django-import-export/django-import-export/blob/main/import_export/results.py#L131) to be changed to namedtuple or similar formatted the same as currently but with expanded info about exact value/column where error happened. 
Caller can than more easily format this and return back in same file by appending it or new file or show in template...

As far i can see this return line [import_obj](https://github.com/django-import-export/django-import-export/blob/main/import_export/resources.py#L543) would need modification.

I would open pr for this. Any hints before i start?

### PR DESCRIPTION
**Pull request by matthewhegarty**
**Problem**

Closes #1359

- Added error information including line number and row (where possible).
- Documented changes
- Added tests and tested manually

### CODE REVIEW

File: docs/advanced_usage.rst [hunk 1]
```diff
@@ -222,16 +222,77 @@ Errors are retained in each :class:`~import_export.results.RowResult` instance w
...
 
     result = self.resource.import_data(self.dataset, raise_errors=False)
-    if result.has_errors():
-        for row in result.rows:
-            for error in row.errors:
-                print(str(error.error))
+    for row in result.invalid_rows:
+        print(f"--- row {row.number} ---")
+        for field, error in row.error.error_dict.items():
+            print(f"{field}: {error}")
```
Comment (matthewhegarty) [inline docs/advanced_usage.rst:256]: Should also illustrate how to output row here. See #1215

Comment (matthewhegarty) [reply docs/advanced_usage.rst]: DONE

File: import_export/resources.py [hunk 2]
```diff
@@ -659,9 +658,11 @@ def handle_import_error(self, result, error, raise_errors=False):
         logger.debug(error, exc_info=error)
         if result:
             tb_info = traceback.format_exc()
-            result.append_base_error(self.get_error_result_class()(error, tb_info))
+            result.append_base_error(
+                self.get_error_result_class()(error, traceback=tb_info)
```
Comment (pokken-magic) [inline import_export/resources.py:662]: this is nice work making it pluggable!

**Review by pokken-magic — APPROVED** [review]
I don't see anythign that jumps out wrong to me about this, looks like great code. Just have to trust the test suite and manual testing :)

**Pull request merged**